Traceback (most recent call last):
  File "c:\users\ishun_000\appdata\local\programs\python\python38\lib\site-packages\jupyter_cache\executors\utils.py", line 58, in single_nb_execution
    executenb(
  File "c:\users\ishun_000\appdata\local\programs\python\python38\lib\site-packages\nbclient\client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "c:\users\ishun_000\appdata\local\programs\python\python38\lib\site-packages\jupyter_core\utils\__init__.py", line 172, in wrapped
    return loop.run_until_complete(inner)
  File "c:\users\ishun_000\appdata\local\programs\python\python38\lib\asyncio\base_events.py", line 616, in run_until_complete
    return future.result()
  File "c:\users\ishun_000\appdata\local\programs\python\python38\lib\site-packages\nbclient\client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "c:\users\ishun_000\appdata\local\programs\python\python38\lib\site-packages\nbclient\client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "c:\users\ishun_000\appdata\local\programs\python\python38\lib\site-packages\nbclient\client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
from Crypto.Util import number
from random import randint

class DHE:
    def __init__(self,k):
        self.q = number.getPrime(k)
        #generator
        self.g=2
    
    def A_key(self):
        self.a = randint(1,self.q-1)
        return pow(self.g,self.a,self.q)
    
    def B_key(self):
        self.b = randint(1,self.q-1)
        return pow(self.g,self.b,self.q)
    
    def Share_key_A(self,yb):
        return pow(yb,self.a,self.q)
    
    def Share_key_B(self,ya):
        return pow(ya,self.b,self.q)
    

dhe = DHE(128)
a_key = dhe.A_key()
b_key = dhe.B_key()

#make shared key by Alice
s_key_A = dhe.Share_key_A(b_key)
#make shared key by Bob
s_key_B = dhe.Share_key_B(a_key)

print("secret key of Alice:",a_key)
print("secret key of Bob:",b_key)
print("Shared key of Alice:",s_key_A)
print("Shared key of Bob",s_key_B)

------------------


[1;31m---------------------------------------------------------------------------[0m
[1;31mModuleNotFoundError[0m                       Traceback (most recent call last)
Cell [1;32mIn[1], line 1[0m
[1;32m----> 1[0m [38;5;28;01mfrom[39;00m [38;5;21;01mCrypto[39;00m[38;5;21;01m.[39;00m[38;5;21;01mUtil[39;00m [38;5;28;01mimport[39;00m number
[0;32m      2[0m [38;5;28;01mfrom[39;00m [38;5;21;01mrandom[39;00m [38;5;28;01mimport[39;00m randint
[0;32m      4[0m [38;5;28;01mclass[39;00m [38;5;21;01mDHE[39;00m:

[1;31mModuleNotFoundError[0m: No module named 'Crypto'

